### PR TITLE
Introduction to Reinforcement Learning (RL)

### PR DESCRIPTION
**Pull request by Bayler**
Fixes #387

#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      

#### Grammar Checks
- [x] Have you spellchecked and grammar checked your article? - **Tool:** [Grammarly](https://grammarly.com)
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical Checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution Guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

### CODE REVIEW

**Comment by Bayler** [comment]
This Fixes #387 

Hi @hectorkambow @cyberShaw 

I have had to close the previous PR on #387 because of conflicting files issue. This is a new PR but of the same revised article Peer Reviewed by @cyberShaw

**Comment by Bayler** [comment]
@cyberShaw I had to close the other PR issue due to a conflicting files issue.

**Comment by hectorkambow** [comment]
Plagiarism check done ✔️

**Comment by hectorkambow** [comment]
@cyberShaw I believe this PR (was an older one you had started to review) 
I believe it is ready for another prem/tech review. 
Please let me know if you would like me to re-assign any review 🚀

**Comment by cyberShaw** [comment]
> @cyberShaw I believe this PR (was an older one you had started to review)
> I believe it is ready for another prem/tech review.
> Please let me know if you would like me to re-assign any review 🚀

I'll do a round of tech review on this one tomorrow 🚀

**Comment by hectorkambow** [comment]
@cyberShaw 
Wanted to get a status report on this PR whenever you have a moment. 
Thank you.

**Comment by cyberShaw** [comment]
> @cyberShaw
> Wanted to get a status report on this PR whenever you have a moment.
> Thank you.

I'll review it today 👍

**Review by cyberShaw — CHANGES_REQUESTED** [review]
This article has been great, @Bayler!
I read through it and found that there were quite a few problems with the links. Most of them weren't working. I fixed them in the commit: c348c5c. Make sure you include them properly the next time.

Also, there were quite a few spelling and grammatical mistakes that I noticed. I ran the article through Grammarly and it raised quite a few red flags. Could you please fix them and we can move on to tech review?
[https://grammarly.com](https://grammarly.com)

File: articles/introduction-to-reinforcement-learning/index.md [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+In RL we have key concepts or terms that we need to define. These include agents, state, action, rewards, and the environment.
+
+1.	**Agent**
+The agent decides on what action to take, given an observation from the environment.
+2.	**State**
+The state represents the decision-making factors under consideration observed by the agent.
+3.	**Action**
+The agent selects an action that may change or affect the state and reward.
+4.	**Reward**
+A reward represents the gains and losses in network performance. The reward is for taking a particular action on a particular state in the previous time instance.
```
Comment (hectorkambow) [inline articles/introduction-to-reinforcement-learning/index.md:43]: This sentence was not clear:
"The reward is for taking a particular action on a particular state in the previous time instance"

The reward is given? 
I think its saying that a reward is given to an agent when they perform a particular action on a particular state in a better (or shorter amount) of time???
Not sure

**Comment by Bayler** [comment]
@hectorkambow is it clearer now?

**Comment by hectorkambow** [comment]
@Bayler  Thank you for the edit I will give it another review now- just FYI - this particular article has been through an extensive review from our peer reviewer as well as myself and [the payout](https://github.com/section-io/engineering-education/blob/master/new_contributors/CONTRIBUTING.md#review-approval--payment-processes) will reflect as such.
🚀 Thank you

**Comment by hectorkambow** [comment]
@Bayler 
Just finishing up with your article - it was a nice article learned quite a bit - I really enjoyed the metaphors and examples to explain the concepts. 

Please be sure to go over my edits to see what can be applied moving forward. Thank you 🚀

**Pull request merged**